ci: Prevent certain checks from running on forked PRs

andrewb1269hg opened the pull request:
**Description**:
Add a check for a forked repo to any workflow file steps which use secrets.* token or default GITHUB_TOKEN.

**Related Issue(s)**:
Fixes #16004

nathanklick commented on .github/workflows/node-flow-fsts-daily-interval-02.yaml:
```diff
@@ -102,7 +102,7 @@ jobs:
     needs:
       - crypto-migration-7n-1c
-    if: ${{ !cancelled() && always() }}
+    if: ${{ !cancelled() && always() && !github.event.workflow_dispatch.repository.fork }}
```
Does not work since `always()` always returns `true` plus has some additional behaviors. Need to change the ordering.

nathanklick commented on .github/workflows/node-flow-build-application.yaml:
```diff
@@ -61,6 +61,7 @@ defaults:
 jobs:
   code:
     name: Code
+    if: ${{ !github.event.workflow_dispatch.repository.fork }}
```
```suggestion
```

nathanklick commented on .github/workflows/flow-node-performance-tests.yaml:
```diff
@@ -77,6 +77,7 @@ jobs:
 
       - name: Check If Release Artifact Exist in Bucket
         id: check-if-exist
+        if: ${{ !github.event.workflow_dispatch.repository.fork }}
```
```suggestion
```

nathanklick commented on .github/workflows/flow-node-performance-tests.yaml:
```diff
@@ -99,6 +100,7 @@ jobs:
 
       - name: Reset Network
         id: reset-network
+        if: ${{ !github.event.workflow_dispatch.repository.fork }}
```
```suggestion
```

nathanklick commented on .github/workflows/flow-node-performance-tests.yaml:
```diff
@@ -115,6 +117,7 @@ jobs:
 
       - name: Restore Old Signed State
         id: restore-old-signed-state
+        if: ${{ !github.event.workflow_dispatch.repository.fork }}
```
```suggestion
```

nathanklick commented on .github/workflows/node-flow-deploy-adhoc-artifact.yaml:
```diff
@@ -46,7 +46,8 @@ defaults:
   release-adhoc:
     name: Release [Adhoc]
-    if: ${{ github.event_name == 'workflow_dispatch' }}
+    if: ${{ github.event_name == 'workflow_dispatch' &&
+            !github.event.workflow_dispatch.repository.fork }}
```
```suggestion
```

nathanklick commented on .github/workflows/node-flow-deploy-preview.yaml:
```diff
@@ -73,6 +73,7 @@ jobs:
 
   deploy-tag:
     name: Deploy Tag
+    if: ${{ !github.event.workflow_dispatch.repository.fork }}
```
```suggestion
```

nathanklick commented on .github/workflows/node-flow-deploy-release-artifact.yaml:
```diff
@@ -74,6 +74,7 @@ jobs:
 
   release-tag:
     name: Release [Tag]
+    if: ${{ !github.event.workflow_run.head_repository.fork }}
```
```suggestion
```

nathanklick commented on .github/workflows/node-flow-deploy-release-artifact.yaml:
```diff
@@ -102,6 +103,7 @@ jobs:
 
   release-branch:
     name: Release [Branch]
+    if: ${{ !github.event.workflow_run.head_repository.fork }}
```
```suggestion
```

nathanklick commented on .github/workflows/node-flow-fsts-custom-regression.yaml:
```diff
@@ -59,6 +59,7 @@ concurrency:
 jobs:
   jrs-panel:
     name: JRS Panel
+    if: ${{ !github.event.workflow_dispatch.repository.fork }}
```
```suggestion
```

andrewb1269hg commented on .github/workflows/flow-node-performance-tests.yaml:
```diff
@@ -77,6 +77,7 @@ jobs:
 
       - name: Check If Release Artifact Exist in Bucket
         id: check-if-exist
+        if: ${{ !github.event.workflow_dispatch.repository.fork }}
```
No longer checking this as workflow_dispatch do not need checks.

andrewb1269hg commented on .github/workflows/flow-pull-request-formatting.yaml:
```diff
@@ -51,6 +51,7 @@ jobs:
           egress-policy: audit
 
       - name: Check PR Title
+        if: ${{ !github.event.pull_request.base.repo.fork }}
```
Done.

andrewb1269hg replied:
This doesn't appear to have been updated. I'm moving it to the level under `title-check` step so it will skip on the whole job.

andrewb1269hg commented on .github/workflows/node-flow-fsts-daily-interval-01.yaml:
```diff
@@ -68,7 +68,7 @@ jobs:
       grafana-agent-username: ${{ secrets.GRAFANA_AGENT_USERNAME }}
       grafana-agent-password: ${{ secrets.GRAFANA_AGENT_PASSWORD }}
-    if: ${{ !cancelled() && always() }}
+    if: ${{ !cancelled() && always() && !github.event.workflow_dispatch.repository.fork }}
```
`workflow_dispatch` for this file, so removing the fork checks.

andrewb1269hg commented on .github/workflows/node-flow-fsts-daily-interval-01.yaml:
```diff
@@ -91,7 +91,7 @@ jobs:
       grafana-agent-username: ${{ secrets.GRAFANA_AGENT_USERNAME }}
       grafana-agent-password: ${{ secrets.GRAFANA_AGENT_PASSWORD }}
-    if: ${{ !cancelled() && always() }}
+    if: ${{ !cancelled() && always() && !github.event.workflow_dispatch.repository.fork }}
```
`workflow_dispatch` for this file, so removing the fork checks.

andrewb1269hg replied:
Updated.

andrewb1269hg commented:
Major updates:
1. Removed all `workflow_dispatch` checks.
2. Added checks to `workflow_call` files.
3. Reformat and standardize the order of if-check stanzas inside steps.

nathanklick commented on .github/workflows/flow-pull-request-formatting.yaml:
```diff
@@ -44,6 +44,7 @@ jobs:
   title-check:
     name: Title Check
     runs-on: network-node-linux-medium
+    if: ${{ !github.event.pull_request.base.repo.fork }}
```
```suggestion
```

nathanklick commented on .github/workflows/node-zxc-build-release-artifact.yaml:
```diff
@@ -373,15 +373,22 @@ jobs:
       - name: Upload Artifacts (DevOps GCP Bucket)
         uses: google-github-actions/upload-cloud-storage@386ab77f37fdf51c0e38b3d229fad286861cc0d0 # v2.2.1
-        if: ${{ inputs.dry-run-enabled != true && !cancelled() && !failure() }}
+        if: ${{ !github.event.workflow_call.repository.fork &&
```
Just disable the whole job.

nathanklick commented on .github/workflows/node-zxc-build-release-artifact.yaml:
```diff
@@ -373,15 +374,20 @@ jobs:
       - name: Upload Artifacts (CDN Bucket)
         uses: google-github-actions/upload-cloud-storage@386ab77f37fdf51c0e38b3d229fad286861cc0d0 # v2.2.1
-        if: ${{ inputs.dry-run-enabled != true && needs.validate.outputs.prerelease != 'true' && !cancelled() && !failure() }}
+        if: ${{ inputs.dry-run-enabled != true &&
+          needs.validate.outputs.prerelease != 'true'&&
+          !cancelled() &&
+          !failure() }}
```
Does this exceed a single line?

andrewb1269hg replied:
I collapsed it into a single line and it looks fine.

nathanklick commented on .github/workflows/node-zxc-build-release-artifact.yaml:
```diff
@@ -532,6 +537,7 @@ jobs:
     needs:
       - validate
       - build-artifact
+    if: ${{ !github.event.workflow_call.repository.fork }}
```
Not needed because the `build-artifact` job is skipped and this workflow requires/needs the `build-artifact` job. Either add as a job level `if` statement for belt and suspenders or remove entirely. Former recommended.

nathanklick commented on .github/workflows/node-zxc-build-release-artifact.yaml:
```diff
@@ -550,6 +556,7 @@ jobs:
     needs:
       - validate
       - build-artifact
+    if: ${{ !github.event.workflow_call.repository.fork }}
```
Same as above

nathanklick commented on .github/workflows/node-zxc-build-release-artifact.yaml:
```diff
@@ -568,7 +575,10 @@ jobs:
       - gcp-production-image
       - jfr-production-image
-    if: ${{ inputs.dry-run-enabled != true && !cancelled() && !failure() }}
+    if: ${{ !github.event.workflow_call.repository.fork &&
+      inputs.dry-run-enabled != true &&
+      !cancelled() &&
+      !failure() }}
```
```suggestion
    if: ${{ inputs.dry-run-enabled != true && !cancelled() && !failure() }}
```

nathanklick commented on .github/workflows/node-zxc-build-release-artifact.yaml:
```diff
@@ -681,7 +691,7 @@ jobs:
     needs:
       - validate
-    if: ${{ inputs.release-profile != 'none' }}
+    if: ${{ !github.event.workflow_call.repository.fork && inputs.release-profile != 'none' }}
```
```suggestion
    if: ${{ inputs.release-profile != 'none' }}
```

nathanklick commented on .github/workflows/node-zxc-build-release-artifact.yaml:
```diff
@@ -863,7 +873,11 @@ jobs:
       - validate-production-image
       - publish
-    if: ${{ inputs.dry-run-enabled != true && inputs.version-policy == 'specified' && !cancelled() && !failure() }}
+    if: ${{ !github.event.workflow_call.repository.fork &&
+      inputs.dry-run-enabled != true &&
+      inputs.version-policy == 'specified' &&
+      !cancelled() &&
+      !failure() }}
```
```suggestion
    if: ${{ inputs.dry-run-enabled != true && inputs.version-policy == 'specified' && !cancelled() && !failure() }}
```

nathanklick commented on .github/workflows/node-zxc-compile-application-code.yaml:
```diff
@@ -160,6 +160,7 @@ jobs:
   compile:
     name: ${{ inputs.custom-job-label || 'Compiles' }}
     runs-on: network-node-linux-large
+    if: ${{ !github.event.workflow_call.repository.fork }}
```
This completely disables compilation checks on all forked PRs and these should run.

andrewb1269hg replied:
Removed this if-check.

rbarkerSL reviewed (dismissed):
LGTM
